proposals/lip-0053.md

```diff
@@ -630,6 +628,13 @@ _Figure 1: A schematic of the steps of a mainchain CCU execution. Some details a
 def execute(trs: Transaction) -> None:
     ccu = trs
     ccu.params = decode(crossChainUpdateTransactionParams, trs.params)
+
+    if ccu.params.inboxUpdate is not empty:
+        # This check is expensive. Therefore, it is done in the execute step instead of the verify
+        # step. Otherwise, a malicious relayer could spam the transaction pool with computationally
+        # costly CCU verifications without paying fees.
+        verifyPartnerChainOutboxRoot(ccu)
```

Comment:
This works, but maybe is not optimal, since the SMT inclusion proof is not necessarily performed in `verifyPartnerChainOutboxRoot`. One option would be to refactor `verifyPartnerChainOutboxRoot` to not perform the expensive check but still the cheap ones, and move only the SMT verify here. @gkoumout Opinions?

Reply:
Agree that this would be better, but it is fine for me to keep it as is for simplicity.

Reply:
Although it does not seem to be a big edit: Just remove lines 280-295 from `verifyPartnerChainOutboxRoot` and paste it here; Note that the name `verifyPartnerChainOutboxRoot` would not make much sense anymore in this case.

Reply:
moving lines 280-295 to execute also requires to move lines 253-263 to execute. And they are also slightly expensive I guess, so it's good to do it in the execute step (then we should also move 298-300).
The only remaining lines that could stay in the verify step are 266-278. Do you agree?

Reply:
I actullay extracted the verifycation of outboxRootWitness into a new function `verifyOutboxRootWitness` and call this one in the verify step.

Reply:
cbr for me